File: src/platforms/react-native/tokens.ts
/**
 * React Native Token Mapping - The Clean Lab
 * "Titanium Command" Industrial/Tactical Design System
 *
 * Token mapping for React Native mobile apps.
 * All values converted to RN-compatible formats.
 */

export const tokens = {
  /* ========================================
     COLORS
     ======================================== */
  colors: {
    // Neutrals
    white: '#ffffff',
    gray50: '#f8fafc',
    gray100: '#f0f2f5',
    gray200: '#e2e8f0',
    gray300: '#cbd5e1',
    gray400: '#94a3b8',
    gray500: '#64748b',
    gray900: '#0f172a',
    black: '#111111',

    // Brand
    blue500: '#0055ff',
    blue600: '#0044cc',
    orange500: '#ff4800',
    orange600: '#e04000',

    // Status
    green500: '#00cc66',
    red500: '#ff3333',
    amber500: '#ffaa00',

    // Semantic - Background
    bgPrimary: '#f0f2f5',
    bgSecondary: '#ffffff',
    bgTertiary: '#f8fafc',

    // Semantic - Text
    textPrimary: '#111111',
    textSecondary: '#64748b',
    textMuted: '#94a3b8',
    textInverse: '#ffffff',

    // Semantic - Accent
    accentPrimary: '#0055ff',
    accentPrimaryHover: '#0044cc',
    accentWash: '#0055ff',
    accentDry: '#ff4800',

    // Semantic - Border
    borderSubtle: '#e2e8f0',
    borderDefault: '#cbd5e1',
    borderStrong: '#94a3b8',

    // Semantic - Status
    statusSuccess: '#00cc66',
    statusSuccessBg: '#e6f9f0',
    statusSuccessText: '#006633',
    statusError: '#ff3333',
    statusErrorBg: '#ffebe6',
    statusErrorText: '#cc0000',
    statusWarning: '#ffaa00',
    statusWarningBg: '#fff8e6',
    statusWarningText: '#cc8800',
  },

  /* ========================================
     SPACING (in pixels for RN)
     ======================================== */
  spacing: {
    0: 0,
    1: 4,
    2: 8,
    3: 12,
    4: 16,
    5: 20,
    6: 24,
    8: 32,
    10: 40,
    12: 48,
    16: 64,
    20: 80,
    24: 96,
    32: 128,
  },

  /* ========================================
     TYPOGRAPHY
     ======================================== */
  typography: {
    // Font families
    fontDisplay: 'Inter',
    fontMono: 'JetBrainsMono',

    // Font sizes (in pixels)
    textXs: 12,
    textSm: 14,
    textBase: 16,
    textLg: 18,
    textXl: 20,
    text2xl: 24,
    text3xl: 30,
    text4xl: 36,

    // Font weights
    weightRegular: '400',
    weightMedium: '500',
    weightSemibold: '600',
    weightBold: '700',
    weightExtrabold: '800',
    weightBlack: '900',

    // Line heights (unitless multiplier for RN)
    leadingTight: 1.1,
    leadingSnug: 1.25,
    leadingNormal: 1.5,
    leadingRelaxed: 1.625,

    // Letter spacing (in pixels)
    trackingTight: -0.4,
    trackingNormal: 0,
    trackingWide: 0.8,
    trackingWidest: 1.6,
  },

  /* ========================================
     SHAPES - Chamfer values for SVG drawing
     ======================================== */
  shapes: {
    // Chamfer sizes (for polygon path calculation)
    chamferSm: 8,
    chamferMd: 12,
    chamferLg: 20,
    chamferXl: 30,

    // Border widths
    borderWidthSm: 1,
    borderWidthMd: 2,
    borderWidthLg: 3,
    borderWidthXl: 4,

    // Border radius (for fallback if SVG not used)
    radiusNone: 0, // Use chamfers instead
  },

  /* ========================================
     SHADOWS (iOS/Android elevation)
     ======================================== */
  shadows: {
    sm: {
      shadowColor: '#000000',
      shadowOffset: { width: 0, height: 1 },
      shadowOpacity: 0.05,
      shadowRadius: 2,
      elevation: 1,
    },
    md: {
      shadowColor: '#000000',
      shadowOffset: { width: 0, height: 4 },
      shadowOpacity: 0.07,
      shadowRadius: 6,
      elevation: 3,
    },
    lg: {
      shadowColor: '#000000',
      shadowOffset: { width: 0, height: 10 },
      shadowOpacity: 0.1,
      shadowRadius: 15,
      elevation: 6,
    },
    xl: {
      shadowColor: '#000000',
      shadowOffset: { width: 0, height: 20 },
      shadowOpacity: 0.15,
      shadowRadius: 25,
      elevation: 10,
    },
  },

  /* ========================================
     TOUCH TARGETS (Mobile-specific)
     ======================================== */
  touch: {
    minTargetSize: 44, // iOS/Android minimum
    recommendedSpacing: 8,
  },

  /* ========================================
     ANIMATIONS
     ======================================== */
  animations: {
    durationFast: 150,
    durationNormal: 200,
    durationSlow: 300,
    easingStandard: 'ease-in-out',
  },
};

/* ========================================
   HELPER FUNCTION: Generate chamfer polygon points
   ======================================== */

export function getChamferPoints(
  width: number,
  height: number,
  chamfer: number
): string {
  return `
    ${chamfer},0
    ${width},0
    ${width},${height - chamfer}
    ${width - chamfer},${height}
    0,${height}
    0,${chamfer}
  `.trim();
}

/* ========================================
   TYPE EXPORTS
   ======================================== */

export type Tokens = typeof tokens;
export type ColorKey = keyof typeof tokens.colors;
export type SpacingKey = keyof typeof tokens.spacing;
export type TypographyKey = keyof typeof tokens.typography;
export type ShapeKey = keyof typeof tokens.shapes;

File: src/platforms/flutter/tokens.dart
/**
 * Flutter Token Mapping - The Clean Lab
 * "Titanium Command" Industrial/Tactical Design System
 *
 * Token mapping for Flutter mobile apps.
 * All values converted to Flutter-compatible formats.
 */

import 'package:flutter/material.dart';

class AppTokens {
  /* ========================================
     COLORS
     ======================================== */

  // Neutrals
  static const Color white = Color(0xFFFFFFFF);
  static const Color gray50 = Color(0xFFF8FAFC);
  static const Color gray100 = Color(0xFFF0F2F5);
  static const Color gray200 = Color(0xFFE2E8F0);
  static const Color gray300 = Color(0xFFCBD5E1);
  static const Color gray400 = Color(0xFF94A3B8);
  static const Color gray500 = Color(0xFF64748B);
  static const Color gray900 = Color(0xFF0F172A);
  static const Color black = Color(0xFF111111);

  // Brand
  static const Color blue500 = Color(0xFF0055FF);
  static const Color blue600 = Color(0xFF0044CC);
  static const Color orange500 = Color(0xFFFF4800);
  static const Color orange600 = Color(0xFFE04000);

  // Status
  static const Color green500 = Color(0xFF00CC66);
  static const Color red500 = Color(0xFFFF3333);
  static const Color amber500 = Color(0xFFFFAA00);

  // Semantic - Background
  static const Color bgPrimary = gray100;
  static const Color bgSecondary = white;
  static const Color bgTertiary = gray50;

  // Semantic - Text
  static const Color textPrimary = black;
  static const Color textSecondary = gray500;
  static const Color textMuted = gray400;
  static const Color textInverse = white;

  // Semantic - Accent
  static const Color accentPrimary = blue500;
  static const Color accentPrimaryHover = blue600;
  static const Color accentWash = blue500;
  static const Color accentDry = orange500;

  // Semantic - Border
  static const Color borderSubtle = gray200;
  static const Color borderDefault = gray300;
  static const Color borderStrong = gray400;

  // Semantic - Status
  static const Color statusSuccess = green500;
  static const Color statusSuccessBg = Color(0xFFE6F9F0);
  static const Color statusSuccessText = Color(0xFF006633);
  static const Color statusError = red500;
  static const Color statusErrorBg = Color(0xFFFFEBE6);
  static const Color statusErrorText = Color(0xFFCC0000);
  static const Color statusWarning = amber500;
  static const Color statusWarningBg = Color(0xFFFFF8E6);
  static const Color statusWarningText = Color(0xFFCC8800);

  /* ========================================
     SPACING
     ======================================== */

  static const double space0 = 0;
  static const double space1 = 4;
  static const double space2 = 8;
  static const double space3 = 12;
  static const double space4 = 16;
  static const double space5 = 20;
  static const double space6 = 24;
  static const double space8 = 32;
  static const double space10 = 40;
  static const double space12 = 48;
  static const double space16 = 64;
  static const double space20 = 80;
  static const double space24 = 96;
  static const double space32 = 128;

  /* ========================================
     TYPOGRAPHY
     ======================================== */

  // Font families
  static const String fontDisplay = 'Inter';
  static const String fontMono = 'JetBrainsMono';

  // Font sizes
  static const double textXs = 12;
  static const double textSm = 14;
  static const double textBase = 16;
  static const double textLg = 18;
  static const double textXl = 20;
  static const double text2xl = 24;
  static const double text3xl = 30;
  static const double text4xl = 36;

  // Font weights
  static const FontWeight weightRegular = FontWeight.w400;
  static const FontWeight weightMedium = FontWeight.w500;
  static const FontWeight weightSemibold = FontWeight.w600;
  static const FontWeight weightBold = FontWeight.w700;
  static const FontWeight weightExtrabold = FontWeight.w800;
  static const FontWeight weightBlack = FontWeight.w900;

  // Line heights (as height multiplier)
  static const double leadingTight = 1.1;
  static const double leadingSnug = 1.25;
  static const double leadingNormal = 1.5;
  static const double leadingRelaxed = 1.625;

  // Letter spacing
  static const double trackingTight = -0.4;
  static const double trackingNormal = 0;
  static const double trackingWide = 0.8;
  static const double trackingWidest = 1.6;

  /* ========================================
     SHAPES
     ======================================== */

  // Chamfer sizes (for custom shape drawing)
  static const double chamferSm = 8;
  static const double chamferMd = 12;
  static const double chamferLg = 20;
  static const double chamferXl = 30;

  // Border widths
  static const double borderWidthSm = 1;
  static const double borderWidthMd = 2;
  static const double borderWidthLg = 3;
  static const double borderWidthXl = 4;

  /* ========================================
     SHADOWS
     ======================================== */

  static const List<BoxShadow> shadowSm = [
    BoxShadow(
      color: Color(0x0D000000), // 5% opacity
      offset: Offset(0, 1),
      blurRadius: 2,
    ),
  ];

  static const List<BoxShadow> shadowMd = [
    BoxShadow(
      color: Color(0x12000000), // 7% opacity
      offset: Offset(0, 4),
      blurRadius: 6,
    ),
  ];

  static const List<BoxShadow> shadowLg = [
    BoxShadow(
      color: Color(0x1A000000), // 10% opacity
      offset: Offset(0, 10),
      blurRadius: 15,
    ),
  ];

  static const List<BoxShadow> shadowXl = [
    BoxShadow(
      color: Color(0x26000000), // 15% opacity
      offset: Offset(0, 20),
      blurRadius: 25,
    ),
  ];

  // Tactical glows
  static const List<BoxShadow> glowBlue = [
    BoxShadow(
      color: Color(0x4D0055FF), // 30% opacity
      blurRadius: 20,
    ),
  ];

  static const List<BoxShadow> glowOrange = [
    BoxShadow(
      color: Color(0x4DFF4800), // 30% opacity
      blurRadius: 20,
    ),
  ];

  /* ========================================
     TOUCH TARGETS
     ======================================== */

  static const double minTargetSize = 44;
  static const double recommendedSpacing = 8;

  /* ========================================
     ANIMATIONS
     ======================================== */

  static const Duration durationFast = Duration(milliseconds: 150);
  static const Duration durationNormal = Duration(milliseconds: 200);
  static const Duration durationSlow = Duration(milliseconds: 300);

  static const Curve easingStandard = Curves.easeInOut;

  /* ========================================
     HELPER: Text Styles
     ======================================== */

  static TextStyle heading1({Color? color}) => TextStyle(
        fontFamily: fontDisplay,
        fontSize: text4xl,
        fontWeight: weightBlack,
        height: leadingTight,
        letterSpacing: trackingTight,
        color: color ?? textPrimary,
      );

  static TextStyle heading2({Color? color}) => TextStyle(
        fontFamily: fontDisplay,
        fontSize: text3xl,
        fontWeight: weightExtrabold,
        height: leadingSnug,
        letterSpacing: trackingTight,
        color: color ?? textPrimary,
      );

  static TextStyle heading3({Color? color}) => TextStyle(
        fontFamily: fontDisplay,
        fontSize: text2xl,
        fontWeight: weightBold,
        height: leadingSnug,
        color: color ?? textPrimary,
      );

  static TextStyle bodyBase({Color? color}) => TextStyle(
        fontFamily: fontDisplay,
        fontSize: textBase,
        fontWeight: weightRegular,
        height: leadingNormal,
        color: color ?? textPrimary,
      );

  static TextStyle bodyLarge({Color? color}) => TextStyle(
        fontFamily: fontDisplay,
        fontSize: textLg,
        fontWeight: weightRegular,
        height: leadingNormal,
        color: color ?? textPrimary,
      );

  static TextStyle label({Color? color}) => TextStyle(
        fontFamily: fontDisplay,
        fontSize: textSm,
        fontWeight: weightSemibold,
        letterSpacing: trackingWide,
        color: color ?? textMuted,
      );

  static TextStyle mono({Color? color, double? fontSize}) => TextStyle(
        fontFamily: fontMono,
        fontSize: fontSize ?? textBase,
        fontWeight: weightMedium,
        color: color ?? textPrimary,
      );
}

/* ========================================
   CUSTOM CHAMFERED BORDER SHAPE
   ======================================== */

class ChamferedBorder extends ShapeBorder {
  final double chamfer;
  final Color borderColor;
  final double borderWidth;

  const ChamferedBorder({
    this.chamfer = 20,
    this.borderColor = Colors.black,
    this.borderWidth = 2,
  });

  @override
  EdgeInsetsGeometry get dimensions => EdgeInsets.all(borderWidth);

  @override
  Path getInnerPath(Rect rect, {TextDirection? textDirection}) {
    return getOuterPath(
      rect.deflate(borderWidth),
      textDirection: textDirection,
    );
  }

  @override
  Path getOuterPath(Rect rect, {TextDirection? textDirection}) {
    return Path()
      ..moveTo(rect.left + chamfer, rect.top)
      ..lineTo(rect.right, rect.top)
      ..lineTo(rect.right, rect.bottom - chamfer)
      ..lineTo(rect.right - chamfer, rect.bottom)
      ..lineTo(rect.left, rect.bottom)
      ..lineTo(rect.left, rect.top + chamfer)
      ..close();
  }

  @override
  void paint(Canvas canvas, Rect rect, {TextDirection? textDirection}) {
    final paint = Paint()
      ..color = borderColor
      ..style = PaintingStyle.stroke
      ..strokeWidth = borderWidth;

    canvas.drawPath(getOuterPath(rect), paint);
  }

  @override
  ShapeBorder scale(double t) {
    return ChamferedBorder(
      chamfer: chamfer * t,
      borderColor: borderColor,
      borderWidth: borderWidth * t,
    );
  }
}
